snippets/src/offerCreate-v2.ts

```diff
@@ -0,0 +1,16 @@
+const client = new WsClient({server: `wss://xrpl.ws`})
+const wallet: Wallet = Wallet.walletFromSeed("sEd…")
+
+const tx: OfferCreateTransaction = {
+    account: "r…",
+    taker_gets: "XRP",
+    taker_pays: {...}
+}
+
+const preparedTransaction = client.autofill(tx)
+const signedTransaction = sign(wallet, preparedTransaction)
```

Comment:
I would prefer to keep the return value the same as the existing `sign` method which returns a tuple of {id, signedTransaction}. For some use cases, the `id` is useful for looking up the tx later. It can be calculated from the signedTransaction, but it's convenient to also have it available upfront.
```suggestion
const signedTransaction = sign(wallet, preparedTransaction).signedTransaction
```

Reply:
Thats a good idea, is the `id` the hash of the Transaction???

```js
const {signedTransaction} = sign(wallet, preparedTransaction)
```

Reply:
Yes. I'd be ok with renaming it to hash if we think that's clearer than id. Not sure what it's called in other libraries.

Reply:
we need more than just `sign`. We should have a module of functions that deal with various signing flows, including for sure as @intelliot mentioned the multisign flow. But! Signing and submitting should be decoupled from each other such that we have:

1) The `client` module/classes, which knows how to do "network things". Users will know that unless it's from that module/class it has no network interaction. This module doesn't know how to sign, but it does know how to autofill and _call out_ to a separate signer function if necessary, and then submit.
2) The `signer` module, which knows how to do a variety of sign actions.

Most users don't want to know about how to deal with this, so the client has a helpful method that just does a single sign, autofill, and submit. But for users with other needs they will sign using functions from the `signer` module and then submit using the `client` module.

What do you think?

Reply:
It's a nice idea. Currently, the client is called RippleAPI, and in v2, I'd like to call it XRPL. I think it would be reasonable to move (more) signing functionality into ripple-keypairs (if desired) and have that effectively be the signer module.